nix/stylish-haskell.nix

```diff
@@ -0,0 +1,16 @@
+{ pkgs
```

Comment:
The current `stylish-haskell` is pulled in here

https://github.com/input-output-hk/ouroboros-network/blob/e142c7153b9ff0505a84ec68124f5bd80449eebd/shell.nix#L30

May we can make the change on that line? If not, we'd at least have to remove that, probably.

I don't know the details of this. Marcin might, or else the devops-crossteam-ci Slack channel eg.

Reply:
Ah, I see now that you removed it below. I suppose setting that to `0.12.2.0` didn't work?

I wonder which part of `haskell.nix` upstream would have to update in order to support just changing that line.

Reply:
modifying it to 0.12.2.0 did not work, those are some "hard-coded" tools within haskell.nix.
it's just easier to point to an existing release

Reply:
Oh, and one more thing, with some things I'm working on atm in stylish we will most likely for a while work on an "unreleased" `0.12.3.0`

Reply:
Looks like `tools` should pick up things that are on Hackage.

https://github.com/input-output-hk/haskell.nix/blob/3f9e7d746895aa9811867a85a3790ba1e3da2798/overlays/tools.nix#L96-L105

Jasper uploaded `0.12.2.0` for the 2020-10-08T12:49:46Z hackage timestamp. The `cabal` file on this branch specifies `index-state: 2020-12-11T00:00:00Z`, which is later. I wonder why that isn't enough for `tools` to find that `stylish-haskell`. :shrug:

Maybe that's a `tools` bug?

Reply:
Actually, looks like it works on `master` as of today (maybe the index-state was increased recently? /shrug).

```
nfrisby@frisbycomp:~/ouroboros-network$ git lg -1
59e71cb54 - (HEAD, origin/master, origin/bors/staging, origin/HEAD) Merge #2965 (4 days ago) <iohk-bors[bot]>
nfrisby@frisbycomp:~/ouroboros-network$ git diff
diff --git a/shell.nix b/shell.nix
index b25df69a9..b70e01dd8 100644
--- a/shell.nix
+++ b/shell.nix
@@ -27,7 +27,7 @@ let
     ];
 
     tools = {
-      stylish-haskell = "0.11.0.0";
+      stylish-haskell = "0.12.2.0";
       ghcid = "0.8.7";
       cabal = "3.2.0.0";
       # todo: add back the build tools which are actually necessary
nfrisby@frisbycomp:~/ouroboros-network$ nix-shell --run 'stylish-haskell --version' 2>&1 | tail -n1
stylish-haskell 0.12.2.0
```